pkg/downloader/chart_downloader.go

```diff
@@ -158,6 +162,71 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+type inMemoryChart struct {
+	ChartData              *bytes.Buffer
+	ProvenanceVerification *provenance.Verification
+}
+
+// inMemoryCharts maps version to a particular chart
+type inMemoryCharts map[string]inMemoryChart
+
+// inMemoryChartList maps a chart name to a particular set of charts
+type inMemoryChartList map[string]inMemoryCharts
+
+var inMemoryCache = make(inMemoryChartList)
```

Comment:
I would be cautious when naming something as cache, and not providing any invalidation/replacement strategy. Currently, this is just a map that will grow indefinitely (certainly bad thing for somebody who wants to run a long-living service)

Reply:
The current implementation works in a content addressable way. Stuff is not added on top of the existing continuously. But I agree that considering that variable here is called cache there needs to be a invalidation/replacement strategy.

Reply:
If I have a service that install various charts over the time this map will just grow in size. I do not see any place in this PR that removes entries from it.